### DISCUSSION TITLE
Use a plugin that's not imported, but required

### DISCUSSION THREAD

**Discussion by Haniver** [remarkjs/remark#985] [Q&A] [answered]
I want to highlight code blocks in my React application using react-remark, but two of the [recommended ](https://github.com/remarkjs/remark/blob/main/doc/plugins.md) plugins use **require**, while react-remark [documentation](https://github.com/remarkjs/react-remark) makes it a must that the plugins are **import**ed. These plugins that are **require**d are:

- [remark-prism](https://github.com/sergioramos/remark-prism)
- [remark-tree-sitter](https://github.com/samlanning/remark-tree-sitter)

The other recommended plugin, which is actually **import**ed, is:
[remark-highlight.js](https://github.com/remarkjs/remark-highlight.js)

And you'd think it would work, but it doesn't ([here's](https://github.com/remarkjs/remark/discussions/984) my post asking for help)

So how do I use **require** to get hold of a plugin that I can later use with react-remark?

**Comment by wooorm** [comment, marked as answer]
- please use https://github.com/rehypejs/rehype-highlight if possible, as documented in the old remark-highlight.js
- those plugins are listed, not recommended
- all maintained plugins have to be imported
- react-remark is out of date I believe, you might want to use react-markdown

**Comment by Haniver** [comment]
Switched to react-markdown as you said, plus a little hack that I describe in my [other post](https://github.com/remarkjs/remark/discussions/984) and now it works!
